components/org.wso2.carbon.identity.api.server.claim.management/org.wso2.carbon.identity.rest.api.server.claim.management.v1/src/main/java/org/wso2/carbon/identity/rest/api/server/claim/management/v1/core/ServerClaimManagementService.java

```diff
@@ -743,6 +942,111 @@ private LocalClaim createLocalClaim(LocalClaimReqDTO localClaimReqDTO) {
+    }
+
+    private ClaimDialectConfiguration getDialectFromFile(InputStream fileInputStream, Attachment fileDetail)
+            throws ClaimMetadataException {
+
+        try {
+            FileContent claimFileContent = new FileContent(fileDetail.getDataHandler().getName(),
+                    fileDetail.getDataHandler().getContentType(),
+                    IOUtils.toString(fileInputStream, StandardCharsets.UTF_8.name()));
+            return generateModelFromFile(claimFileContent);
+        } catch (IOException | ClaimMetadataClientException e) {
+            throw new ClaimMetadataException(ERROR_CODE_INVALID_INPUT_FILE.toString(), e);
+        } finally {
+            IOUtils.closeQuietly(fileInputStream);
+        }
+    }
+
+    private ClaimDialectConfiguration generateModelFromFile(FileContent fileContent) throws ClaimMetadataException {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug(String.format("Parsing Claim Dialect from file: %s of type: %s.", fileContent.getFileName(),
+                    fileContent.getFileType()));
+        }
+        if (StringUtils.isEmpty(fileContent.getContent())) {
+            throw handleClaimManagementClientError(ERROR_CODE_MISSING_FILE_CONTENT, BAD_REQUEST,
+                    fileContent.getFileName());
+        }
+
+        switch (Util.getMediaType(fileContent.getFileType())) {
+            case MEDIA_TYPE_XML:
+                return parseClaimFromXml(fileContent);
+            case MEDIA_TYPE_JSON:
+                return parseClaimFromJson(fileContent);
+            case MEDIA_TYPE_YAML:
+                return parseClaimFromYaml(fileContent);
+            default:
+                LOG.warn(String.format("Unsupported media type %s for file %s. Defaulting to YAML parsing.",
+                        fileContent.getFileType(), fileContent.getFileName()));
+                return parseClaimFromYaml(fileContent);
+        }
+    }
+
+    private ClaimDialectConfiguration parseClaimFromXml(FileContent fileContent) throws ClaimMetadataException {
+
+        try {
+            JAXBContext jaxbContext = JAXBContext.newInstance(ClaimDialectConfiguration.class);
+            Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
+            return (ClaimDialectConfiguration) unmarshaller.unmarshal(new StringReader(fileContent.getContent()));
+        } catch (JAXBException e) {
+            throw new ClaimMetadataException(String.format(ERROR_CODE_ERROR_READING_FILE_CONTENT.toString(),
+                    MEDIA_TYPE_XML), e);
+        }
+    }
+
+    private ClaimDialectConfiguration parseClaimFromJson(FileContent fileContent) throws ClaimMetadataException {
+
+        try {
+            return new ObjectMapper().readValue(fileContent.getContent(), ClaimDialectConfiguration.class);
+        } catch (JsonProcessingException e) {
+            throw new ClaimMetadataException(String.format(ERROR_CODE_ERROR_READING_FILE_CONTENT.toString(),
+                    MEDIA_TYPE_JSON), e);
```

Comment:
Why convert the enum to a string here. We should get the message using the enum here.

Reply:
Check other places as well